app/services/site_settings_task.rb

```diff
@@ -38,4 +38,35 @@ def self.import(yml)
     end
     [log, counts]
   end
+
+  def self.names
+    site_settings = SiteSetting.all_settings(include_hidden: true)
+    site_settings.map { |site_setting| site_setting[:setting].to_s }
+  end
+
+  def self.rg_installed?
+    !`which rg`.strip.empty?
+  end
+
+  def self.directory_path(directory_name)
+    File.expand_path(File.join(Dir.pwd, "..", directory_name))
+  end
+
+  def self.directories_to_check
+    %w[all-the-themes all-the-custom-themes all-the-plugins all-the-custom-plugins]
```

Comment:
Only thing I would add is being able to specify a directory (using an ENV) where all these repos are kept...I keep these totally separate from the core discourse repo in `$HOME/allthe`

Reply:
Ya everyone's setups are different, good call, I'll add that.